common/scala/src/main/scala/whisk/core/entity/InstanceId.scala

```diff
@@ -22,7 +22,20 @@ import spray.json.DefaultJsonProtocol
 case class InstanceId(val instance: Int, name: Option[String] = None) {
   def toInt: Int = instance
 }
-
+case class StringInstanceId(instance: String) {
```

Comment:
It might be worth considering naming this `ControllerInstanceId`? (and `InstanceId` -> `InvokerInstanceId`)
WDYT?

Reply:
is there a reason to distinguish the two - in other words, could they both be StringInstanceId? (in which case can rename StringInstanceId to InstanceId)

Reply:
Currently InvokerSupervision does update by index - where the index is directly mapped from the instance id using:
```
status = status.updated(instance.toInt, new InvokerHealth(instance, currentState))
```
I think this would have to change to use a `Map` instead of `IndexedSeq`

Reply:
We could preserve the duality with toInt returning the hashcode of the string?

Reply:
A map may be fine as well...

Reply:
If it's a map, we can get rid of the invoker `--name` arg, I think? @dgrove-oss WDYT? If the instance id is always a string for invoker, is there a need to map the `--name` arg to an int anymore?

Reply:
can use a `ListMap` to preserve order of invoker ids - I believe the current loadbalancer is sensitive to the preserved order of invokers during scheduling

Reply:
Need some ideas here - the current invoker list is an `IndexesSeq` in loadbalancer, and the stepping behavior relies on the order being preserved even while invokers are added with gaps in the ids. 
If using a `Map` etc, this will break the stepping logic, since adding elements will change the result of stepping to next key. 

This was why I didn't change it originally...

BTW using hashcode could work, except that there is a possibility of collisions. That seems negligible, but possible - we could simply reject any invoker that is added with a colliding hashcode, but this seems bad form.

I'm back to thinking this should be a separate instanceId for controller - `StringInstanceId`, or rename it to `ControllerInstanceId`

Reply:
OK - incremental works. I'll defer to @markusthoemmes on the invoker list.

Reply:
Thanks for trying @tysonnorris.

Reply:
Sorry, late to the party.

I agree we still need both --name and --id independent of whether they are of type String or Int unless we want to make some deeper changes in the loadbalancer.  Might be nice to try to simplify at some point, but I don't think it is worth holding up this PR to accomplish it.